### ISSUE TITLE
hrProcessorLoad = No Such Instance currently exists at this OID, takes 1 minute to fix

### ISSUE THREAD

**Issue by BloodyIron** [net-snmp/net-snmp#896]
Just installed some new Ubuntu Server 24.04 VMs rocking snmpd (net-snmp) v5.9.4 and I'm seeing a new problem, probably related to Linux kernel v6.7 (or v6.8?) as per another thread I found here.

The running kernel is:
6.8.0-48-generic #48-Ubuntu SMP PREEMPT_DYNAMIC Fri Sep 27 14:04:52 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

Whenever snmpd restarts (by telling the daemon to restart, or the OS to restart) it takes about a minute to get actual CPU metrics out of net-snmp.

During this window, an snmpbulkwalk seeking the specific OID info of "hrProcessorLoad" produces this output:
hrProcessorLoad = No Such Instance currently exists at this OID

However, after roughly 1 minute, trying again I get actual output:
hrProcessorLoad.196608 = 1
hrProcessorLoad.196609 = 4
hrProcessorLoad.196610 = 8
hrProcessorLoad.196611 = 1

So far I seem to be the first person to observe this, and I don't think I'm getting this same effect for other Linuxy things I have which are rocking lower Linux Kernel versions.

Anyone have any hot tips as for what I can do about this? This seems like a bug to me.

**Comment by BloodyIron** [comment]
Hmmm I'm not sure if this is actually relevant to Linux Kernel v6.7 or whatever. I think I'm seeing this on other Linux hosts with even kernel v5.15 or similar. Unclear exactly where this is coming from.

**Comment by si458** [comment]
im amazed after 2 years nobody has picked up on this?
i have been bashing my head for months over this,
every so often my processors in librenms would vanish, then on a discovery run 5 mins later they would reappear,
but what was happening was i was adding a snmp_extend service in like of_updates for example,
restarting the snmpd, then going into librenms and discovery again, and it would find the extend but the processors would disappear/vanish!
but indeed if i wait 60 seconds then try discovery again, they would just reappear!

**Comment by bvanassche** [comment]
Is there anything unusual about your configuration like e.g. a very large number of processes, a very large number of storage devices or a very large number of network interfaces?

**Comment by si458** [comment]
No, it happens even on a basic plain ubuntu server 26.04 setup with just snmpd installed
